src/Tools/dotnet-monitor/CollectionRules/ExecuteAction.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules
+{
+    internal sealed class ExecuteAction : IAction<ExecuteOptions>
+    {
+        public async Task<ActionResponse> Execute(ExecuteOptions options, CancellationToken token)
+        {
+            ActionResponse executeResponse = new();
+
+            await Task.Run(() =>
+            {
+                string path = options.Path;
+                string arguments = options.Arguments;
+
+                Process process = Process.Start(path, arguments);
```

Comment:
We might want to capture stdout and stderr and possibly return those in the result, but I don't have a good scenario in mind where we need this. So maybe don't do this and just add a consideration comment in the code to that effect.